python/cudf/cudf/_lib/pylibcudf/null_mask.pyx

```diff
@@ -0,0 +1,157 @@
+
+    Parameters
+    ----------
+    number_of_bits : size_type
+        The number of bits that need to be represented
+
+    Returns
+    -------
+    size_t
+        The necessary number of bytes
+    """
+    cdef size_t output_size
+
+    with nogil:
+        output_size = cpp_null_mask.bitmask_allocation_size_bytes(number_of_bits)
+
+    return output_size
+
+
+cpdef DeviceBuffer create_null_mask(
+    size_type size,
+    mask_state state = mask_state.UNINITIALIZED
+):
+    """Creates a ``DeviceBuffer`` for use as a null value indicator bitmask of a
+    ``Column``.
+
+    For details, see :cpp:func:`create_null_mask`.
+
+    Parameters
+    ----------
+    size : size_type
+        The number of elements to be represented by the mask
+    state : mask_state, optional
+        The desired state of the mask. Can be one of { MaskState.UNALLOCATED,
+        MaskState.UNINITIALIZED, MaskState.ALL_VALID, MaskState.ALL_NULL }
+        (default MaskState.UNINITIALIZED)
+
+    Returns
+    -------
+    rmm.DeviceBuffer
+        A ``DeviceBuffer`` for use as a null bitmask satisfying the desired size and
+        state
+    """
+    cdef device_buffer db
+    cdef unique_ptr[device_buffer] up_db
+
+    with nogil:
+        db = move(cpp_null_mask.create_null_mask(size, state))
+        up_db = move(make_unique[device_buffer](move(db)))
+
+    return DeviceBuffer.c_from_unique_ptr(move(up_db))
+
+
+cpdef tuple bitmask_and(list columns):
+    """Performs bitwise AND of the bitmasks of a list of columns.
+
+    For details, see :cpp:func:`bitmask_and`.
+
+    Parameters
+    ----------
+    columns : list
+        The list of columns
+
+    Returns
+    -------
+    tuple[DeviceBuffer, size_type]
+        A tuple of the resulting mask and count of unset bits
+    """
+    cdef Table c_table = Table(columns)
+    cdef pair[device_buffer, size_type] c_result
+    cdef unique_ptr[device_buffer] up_db
+
+    with nogil:
+        c_result = move(cpp_null_mask.bitmask_and(c_table.view()))
+        up_db = move(make_unique[device_buffer](move(c_result.first)))
+
+    dbuf = DeviceBuffer.c_from_unique_ptr(move(up_db))
+
+    return dbuf, c_result.second
+
+
+cpdef tuple bitmask_or(list columns):
+    """Performs bitwise OR of the bitmasks of a list of columns.
+
+    For details, see :cpp:func:`bitmask_or`.
+
+    Parameters
+    ----------
+    columns : list
+        The list of columns
+
+    Returns
+    -------
+    tuple[DeviceBuffer, size_type]
+        A tuple of the resulting mask and count of unset bits
+    """
+    cdef Table c_table = Table(columns)
+    cdef pair[device_buffer, size_type] c_result
+    cdef unique_ptr[device_buffer] up_db
+
+    with nogil:
+        c_result = move(cpp_null_mask.bitmask_or(c_table.view()))
+        up_db = move(make_unique[device_buffer](move(c_result.first)))
+
+    dbuf = DeviceBuffer.c_from_unique_ptr(move(up_db))
+
+    return dbuf, c_result.second
```

Comment:
It's unfortunate that the rmm object requires a `unique_ptr` and is forcing us to wrap the buffers, but since we have to do it can we wrap this into a helper function? I'm envisioning something like this (not tested):
```
cdef DeviceBuffer buffer_to_python(device_buffer buf):
    return DeviceBuffer.c_from_unique_ptr(move(make_unique[device_buffer](buf)))

# Usage
...
with nogil:
    c_result = ...
return buffer_to_python(move(c_result.first)), c_result.second
```
That can then be reused in all of the functions above.